### PR TITLE
206 restrict a user in a particular user group from performing or accessing anything

### PR DESCRIPTION
**Pull request by longshuicy**
Ready to be reviewed:

This PR catches all the 403 error (which is the backend response when a user doesn't belong to any group) and redirect to a forbidden page; forbids user from any actions.

----

To Test:
1. npm run start:dev
2. Login with **username: testuser2, password: {contact developer}** --> this is a special user that doesn't belong to any group
3. Try to go to any viewers and profile page would direct to the forbidden page 
4. Landing page stays the same

image

### CODE REVIEW

**Comment by Rashmil-1999** [comment]
When I try to go to Jupyterhub, it still gives me this blank screen error.

**Comment by longshuicy** [comment]
> When I try to go to Jupyterhub, it still gives me this blank screen error.

I deployed this to dev. On dev, when you go to jupyterhub with a unauthorized user, the jupyter hub gives below error
image

But this part is related to Jupyter's authenticator; frontend don't have much to do with it.

**Comment by Rashmil-1999** [comment]
Ahh I see... Let me review it ... :)

File: src/components/Forbidden.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+				justifyContent="center"
+				style={{ minHeight: "50vh"}}
+			>
+				<Grid item xs={4}>
+					<Typography variant="h3" paragraph>
+						Access Denied
+					</Typography>
+					<Typography variant="body1" paragraph style={{fontSize: "16px"}}>
+						You do not currently have permission to access the IN-CORE resource. <br />
+						Please wait for the administrator to grant you access.
```
Comment (navarroc) [inline src/components/Forbidden.jsx:21]: I might suggest a minor rewording to "Please wait for an administrator to grant you access".

Comment (longshuicy) [reply src/components/Forbidden.jsx:21]: ok will do!

**Review by Rashmil-1999 — APPROVED** [review]
Looks good

**Review by navarroc — APPROVED** [review]

**Pull request merged**